### PR TITLE
Updated blog model to be extensible

### PR DESCRIPTION
**Pull request by an0o0nym**
Added possibility to extend blog model.

### CODE REVIEW

**Comment by an0o0nym** [comment]
Hi @auvipy  did you mean backwards compatibility for Python2?

**Comment by auvipy** [comment]
no I am talking about the backward compatibility for existing model which is not abstract

**Comment by auvipy** [comment]
introducing abs might introduce bc in migrations

**Comment by an0o0nym** [comment]
@auvipy Ok I see. Are you able to guide me with that one? Cause I am not sure how to do this.

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,10 @@
 Changelog
```
Comment (marctc) [inline CHANGELOG]: This is not necessary. For the moment, this is my job :)

Comment (an0o0nym) [reply CHANGELOG]: Noted!

**Comment by marctc** [comment]
@an0o0nym I'm planning to release puput version 1.0 and I think that this feature could be useful. Are you able to review this PR and make the proper changes? thanks

**Comment by auvipy** [comment]
you can take inspiration about how bc is kept while moving to abstract base model https://github.com/justquick/django-activity-stream/pull/309

**Comment by an0o0nym** [comment]
@marctc @auvipy Thanks for the inspiration. I will try to review it and apply proper changes.

**Comment by an0o0nym** [comment]
@auvipy you wrote:

> no I am talking about the backward compatibility for existing model which is not abstract

But the `BlogPage` is not abstract. This is not the same case as in the link provided in your previous message.

I have just tested my changes manually by installing puput, populating DB with BlogPage entry, applying my changes, applying new migrations and it all works just fine!

**Comment by lukat33** [comment]
When will this be merged to the package?

**Comment by MiltonLn** [comment]
@an0o0nym What's the status of this PR? have you implemented the last changes to it? I would like to have it in our next release since this is pretty useful.

**Comment by an0o0nym** [comment]
This PR mimics the Extending `puput.EntryAbstract` behaviour, which is to let users define their own Entry Abstract model by inheriting from `puput.EntryAbstract` model.

This PR does not change the behaviour of `puput.BlogPage` model. `puput.BlogPage` model is NOT abstract, as @auvipy suggested. Thus it does not need the `swappable` feature to be implemented. 

In my opinion, the swappable feature would qualify for separate PR, where one could define own Blog Page model and swap it for the Puput's `BlogPage` model. 

Purpose of this PR is to allow extending ABSTRACT `BlogAbstract` model, NOT `BlogPage` model itself.

**Comment by marctc** [comment]
@an0o0nym do you wanna still merge this?

**Comment by an0o0nym** [comment]
@marctc I think this question should not be addressed to me :) I am pretty happy with my commit. The question is - are you? I do not see any backward incompatibilities as @auvipy suggested.

So whenever you are ready feel free to merge this branch.

**Comment by marctc** [comment]
Thanks @an0o0nym, do you mind to rebase with master and fix the conflicts? We can add this to next release

**Comment by an0o0nym** [comment]
Sure, I do that tomorrow at latest :+1:

**Comment by an0o0nym** [comment]
@marctc Conflicts resolved.

**Pull request merged**